libsplinter/src/migrations/diesel/sqlite/migrations/2021-02-09-042200_update-identity-enum/up.sql

```diff
@@ -0,0 +1,31 @@
+-- Copyright 2018-2021 Cargill Incorporated
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+-- -----------------------------------------------------------------------------
+
+ALTER TABLE identities RENAME TO _identities_old;
+
+CREATE TABLE identities (
+    identity      TEXT PRIMARY KEY,
+    identity_type TEXT CHECK( identity_type IN ('key','user') ) NOT NULL
+);
+
+INSERT INTO identities (identity, identity_type)
+  SELECT identity, (case
+                      when 1 then 'key'
+                      when 2 then 'user'
+                      else 'unknown'
```

Comment:
If "unknown" was ever set what would happen?

Reply:
Given this code isn't released, I'm not sure that it would matter, but the migration will fail.  I think I will change this to default to 'user', as that currently can be an arbitrary value